### PR TITLE
Pad slot value on comparison

### LINKED ISSUE #769: Pad slot value on comparison
### Description

The current implementation is accepting an `0x` as slot value, which should be converted to `0x0000000000000000000000000000000000000000000000000000000000000000` when compared with the DB value.

### Steps to reproduce

Pass "0x" as slot value on `eth_getStorageAt` endpoint
```
{
  "method": "eth_getStorageAt",
  "params": [
    "0.0.1058",
    "0x",
    "latest"
  ],
  "id": 2,
  "jsonrpc": "2.0"
}
```

### Additional context

_No response_

### Hedera network

other

### Version

latest

### Operating system

None

### PR DESCRIPTION
**Pull request by natanasow**
Signed-off-by: nikolay <n.atanasow94@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #769

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -1127,6 +1127,10 @@ export class EthImpl implements Eth {
     return tag == null || tag === EthImpl.blockLatest || tag === EthImpl.blockPending;
   }
 
+  private static toHex32Byte(input: string): string {
+    return input.length === 66 ? input : EthImpl.emptyHex + this.prune0x(input).padStart(64, '0');
```
Comment (dimitrovmaksim) [inline packages/relay/src/lib/eth.ts:1131]: Instead of hardcoding `64` here I would go with `64 - this.prune0x(input).length`

Comment (dimitrovmaksim) [reply packages/relay/src/lib/eth.ts:1131]: Yeah you're right, sorry, I misunderstood `padStart`

File: packages/relay/src/lib/eth.ts [hunk 2]
```diff
@@ -504,9 +504,11 @@ export class EthImpl implements Eth {
           }
           if (EthImpl.isArrayNonEmpty(contractResultDetails.state_changes)) {
             // filter the state changes to match slot and return value
-            const stateChange = contractResultDetails.state_changes.find(stateChange => stateChange.slot === slot);
-            result = stateChange.value_written;
-          } 
+            const stateChange = contractResultDetails.state_changes.find(stateChange => stateChange.slot === EthImpl.toHex32Byte(slot));
+            if (stateChange) {
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:508]: Update `eth.spec.ts` with coverage for result setting and different paths within `toHex32Byte()`

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Missing coverage

**Review by georgi-l95 — DISMISSED** [review]
LGTM

**Review by Nana-EC — DISMISSED** [review]

**Review by lukelee-sl — DISMISSED** [review]
LGTM

**Review by Nana-EC — DISMISSED** [review]

**Comment by dimitrovmaksim** [comment]
I think the mirror-node images should be updated to 0.70.0 in order for the getStorageAt test to pass, because 0.69.0 does not include this PR https://github.com/hashgraph/hedera-mirror-node/pull/4881

**Review by georgi-l95 — APPROVED** [review]
LGTM

**Review by dimitrovmaksim — APPROVED** [review]

**Pull request merged**